rfcs/0004-enhanced-limits-and-capabilities.md

```diff
@@ -0,0 +1,655 @@
+
+steps:
+- name: StepOne
+  script:
+    actions:
+      onRun:
+        command: python
+        args: [ "-c", "print('This is StepOne - it is light on resources')" ]
+  hostRequirements:
+    amounts:
+    - name: amount.worker.vcpu
+      min: "{{Param.CpuMinAmount}}"
+      max: "{{Param.CpuMaxAmount}}"
+    - name: amount.worker.memory
+      min: "{{Param.MemoryMinAmount}}"
+      max: "{{Param.MemoryMaxAmount}}"
+```
+
+#### notifyPeriodInSeconds Property in \<CancelationMethodNotifyThenTerminate\>
+
+```yaml
+parameterDefinitions:
+- name: NotifyPeriod
+  type: INT
+  minValue: 1
+  maxValue: 600
+  default: 120
+  description: How long to wait for a graceful shutdown before a terminal signal is sent
+
+steps:
+- name: PythonScript
+  script:
+    actions:
+      onRun:
+        command: python
+        args: ['{{Task.File.Run}}']
+        cancelation:
+          mode: NOTIFY_THEN_TERMINATE
+          notifyPeriodInSeconds: "{{Param.NotifyPeriod}}"
+```
+
+### EOL Control in Embedded Files
+
+```yaml
+steps:
+- name: RunBashOnWindows
+  hostRequirements:
+    attributes:
+    - name: attr.worker.os.family
+      anyOf:
+      - windows
+  script:
+    actions:
+      onRun:
+        args:
+        - '{{ Task.File.run }}'
+        command: bash
+      embeddedFiles:
+      - name: run
+        type: TEXT
+        runnable: true
+        data: |
+          #!/bin/bash
+          echo 'Running'
+        endOfLine: LF
+```
+
+### Script Interpreter Syntax Sugar
+
+```yaml
+steps:
+- name: RunPythonScript
+  script:
+    actions:
+      onRun:
+        python: |
+          print('Hello from Python!')
+```
+
+Compare to
+
+```yaml
+steps:
+- name: RunPythonScript
+  script:
+    actions:
+      onRun:
+        command: python
+        args: ["{{Task.File.hello}}"]
+    embeddedFiles:
+      - name: hello
+        filename: "hello.py"
+        type: TEXT
+        data: |
+          print('Hello from Python!')
+```
+
+## Motivation
+
+These enhancements address real-world limitations encountered in production
+OpenJD deployments. The current specification imposes
+constraints that have led to friction when applications require
+extensive parameterization, cross-platform script compatibility, and simplified
+workflow authoring.
+
+These changes are bundled together as creating separate RFCs for each
+of these improvements would be disproportionate to their individual scope.
+Each addresses issues identified through real-world usage,
+and while individually minor, together they improve the
+developer experience and operational flexibility of OpenJD templates.
+
+### Use Cases Supported
+
+1. **Automated Naming**: When a template is generated programmatically, it's
+natural to concatenate identifiers from the input to get a unique name. The
+current length limits prevent that from working naturally. Identifier names can
+similarly be generated by concatenating identifiers, so we likely want to treat
```

Comment:
Where you say "we likely want", you need to make a choice so the proposal is concrete. Either the RFC is making that change too, or it's not.